spatialmath/orientationVector.go

```diff
@@ -141,12 +114,38 @@ func (ov *OrientationVector) Normalize() {
 
 // EulerAngles returns orientation in Euler angle representation.
 func (ov *OrientationVector) EulerAngles() *EulerAngles {
-	return QuatToEulerAngles(ov.ToQuat())
+	return QuatToEulerAngles(ov.Quaternion())
 }
 
 // Quaternion returns orientation in quaternion representation.
 func (ov *OrientationVector) Quaternion() quat.Number {
-	return ov.ToQuat()
+	// make sure OrientationVector is normalized first
+	ov.Normalize()
+
+	// acos(rz) ranges from 0 (north pole) to pi (south pole)
+	lat := math.Acos(ov.OZ)
+
+	// If we're pointing at the Z axis then lon is 0, theta is the OV theta
+	// Euler angles are gimbal locked here but OV allows us to have smooth(er) movement
+	// Since euler angles are used to represent a single orientation, but not to move between different ones, this is OK
+	lon := 0.0
+	theta := ov.Theta
+
+	if 1-math.Abs(ov.OZ) > defaultAngleEpsilon {
+		// If we are not at a pole, we need the longitude
```

Comment:
The vector component of an orientation vector can be thought of as lon/lat as well. This isn't a new change here, this simply moves the contents of `ToQuat` unchanged into `Quaternion()` since there was a useless extra function call.